Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -83,6 +84,31 @@ MatrixWorkspace::~MatrixWorkspace() {
   }
 }
 
+void MatrixWorkspace::setIndexTranslator(
```

Comment:
I think we should decide this based on my question below, regarding cases where only spectrum numbers and not detector IDs get updated.